core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1138,132 @@ def delete_playthroughs_multi(cls, playthrough_ids):
         cls.delete_multi(instances)
 
 
+class LearnerAnswerDetailsModel(base_models.BaseModel):
+    """Model for storing the answer details, learner enters when they
+    are asked for it.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # ID of the entity.
+    entity_id = ndb.StringProperty(required=True, indexed=True)
+    # The type of entity, that whether it is exploration, question, or any
+    # other object if it is added in future.
+    entity_type = ndb.StringProperty(required=True, indexed=True)
+    # The answer submitted by the learner.
+    learner_answer = ndb.StringProperty(required=True, indexed=False)
```

Comment:
Hmm, it would be linked to the "answer object" of an interaction, right? So, yeah, it should probably be json property, but could be some existing classes like maybe RuleSpec? (Don't know which class is used to store the answer)

Reply:
Yeah, learner answer should be a JsonProperty and this need to be versioned.

Also, I see similar models, domain objects used in stats (StateAnswers model/object stores metadata and list of SubmittedAnswer object) which are also versioned.

Reply:
I found that we use to store the answer in string format ([here](https://github.com/oppia/oppia/blob/develop/core/domain/stats_domain.py#L986)), I'm not sure whether that's what we are going to save here?

Reply:
I think that loses information; storing the JsonProperty would probably be better.

/cc @brianrodri for info

Reply:
Hi @seanlip, as I am using JsonProperty of answers submitted by the learners, I checked that in StateAnswers model we store the submitted_answers, which is actually a domain object which consists of mainly rule_spec_index, answer_group_index, interaction_index, interaction_id and many other fields.
So I can similarly make a domain object which will have attributes as
- answer (submitted by the learner: StringProperty)
- interaction_id
- rule_spec_index
- answer_group_index
which will be necessary to perform queries on the submitted answers. And all these fields will be stored in the JSON in the learner_answer field.
What do you suggest?

Reply:
Don't know -- you'll need to analyze it based on the queries you need to perform to get the data for the frontend. Choose a data structure for the backend to ensure that you can perform each query efficiently. Note that stuff in JSON fields can't be queried, and that get_by_id operations are fastest and should be preferred if possible.

You might want to talk to @aks681 or @DubeySandeep for advice on data modelling. I suggest that you also create a short doc that explains what queries you need to perform (input + expected output), since having all those clearly specified in one place will help with figuring out a good data representation. I'm happy to review that if you like.

Reply:
Okay, @seanlip I will make a short doc which will include all the details of the model and share with you.

Reply:
Also @seanlip if learner submits any answer it is of string or json property?

Reply:
Treat it as a JSON property, you need to do something generalizable across multiple types of interaction IDs.

Reply:
Done